op-node: Make sealing duration value configurable

trianglesphere opened the pull request:
**Description**

This makes the sealing duration value configuration via flags.

trianglesphere committed:
op-node: Make sealing duration value configurable

This makes the sealing duration value configuration via flags.

protolambda committed:
op-node: seq seal duration minimum and default value, comment config

protolambda commented:
@trianglesphere 

I added a commit that makes sure there is no negative value, and that 0 automatically becomes the 50ms. There were many instantiations of the op-node config (and indirectly the driver sequencer config) across the testing that were broken otherwise.
If you need an extremely low sequencer sealing duration, you can always configure it explicitly to 1ms. Or even 1ns.
And I added a doc comment on the config field.

What do you think of these changes? And what do you intend to configure it to?

protolambda commented:
/ci authorize 3ebd18bbbc4e3b9a9cd3bd541f9855790b6766ba

trianglesphere commented:
> @trianglesphere 
> 
> 
> 
> I added a commit that makes sure there is no negative value, and that 0 automatically becomes the 50ms. There were many instantiations of the op-node config (and indirectly the driver sequencer config) across the testing that were broken otherwise.
> 
> If you need an extremely low sequencer sealing duration, you can always configure it explicitly to 1ms. Or even 1ns.
> 
> And I added a doc comment on the config field.
> 
> 
> 
> What do you think of these changes? And what do you intend to configure it to?
> 
> 

Those changes make sense. I didn't realize how many times the struct was created elsewhere. Our intention is to increase the duration, but we're not sure on the exact value yet. We don't think that we would benefit from lowering it.

opgitgovernance commented:
This pr has been automatically marked as stale and will be closed in 5 days if no updates

trianglesphere committed:
Merge branch 'ethereum-optimism:develop' into variable-fetch-delay

trianglesphere commented:
@protolambda I think this needs another bump to CI
